scrnaseq.Rmd

```diff
@@ -570,42 +580,112 @@ p2 = PlotMystyle(p2)
 p = p1 + p2 & theme(legend.position="bottom")
 p = p + patchwork::plot_annotation(title="Cell cycle phases")
 p
+```
 
+### UMAP coloured by cell cycle phases
+```{r cellCycleEffect_umap_phases}
 # UMAP with phases superimposed
 p = Seurat::DimPlot(sc, group.by="Phase", pt.size=1)
 PlotMystyle(p, title="UMAP, cells coloured by cell cycle phases", legend_title="Phase")
+```
 
+### UMAP coloured by S phase
+```{r cellCycleEffect_umap_s}
+p = Seurat::FeaturePlot(sc, features="S.Score", pt.size=1, min.cutoff="q1", max.cutoff="q99", cols=c("lightgrey", param$col))
+p = PlotMystyle(p, title="UMAP, cells coloured by S phase")
+p
 ```
 
-```{r cellCycleEffect_featurePlot, fig.height=10}
-p1 = Seurat::FeaturePlot(sc, features="S.Score", pt.size=1, min.cutoff="q1", max.cutoff="q99", cols=c("lightgrey", param$col))
-p1 = PlotMystyle(p1, title="UMAP, cells coloured by S phase")
-p2 = Seurat::FeaturePlot(sc, features="G2M.Score", pt.size=1, min.cutoff="q1", max.cutoff="q99", cols=c("lightgrey", param$col))
-p2 = PlotMystyle(p2, title="UMAP, cells coloured by G2M phase")
-p = p1 + p2
-p 
+### UMAP coloured by G2/M phase
+```{r cellCycleEffect_umap_g2m}
+p = Seurat::FeaturePlot(sc, features="G2M.Score", pt.size=1, min.cutoff="q1", max.cutoff="q99", cols=c("lightgrey", param$col))
+p = PlotMystyle(p, title="UMAP, cells coloured by G2M phase")
+p
 ```
 
-## Feature plots QC
+## Feature plots QC  {.tabset}
 Do cells in individual clusters have particularly high counts, detected genes or mitochondrial content?
-```{r featurePlot_QC, fig.height=10}
-p = Seurat::FeaturePlot(sc, features=features, cols=c("lightgrey", param$col), combine=FALSE)
-names(p) = features
-for (i in features) p[[i]] = PlotMystyle(p[[i]], title=i)
-patchwork::wrap_plots(p, ncol=2)
+
+### Number of counts
+```{r featurePlot_QC_nCount}
+p = Seurat::FeaturePlot(sc, features="nCount_RNA", cols=c("lightgrey", param$col))
```

Comment:
Can we think about other colors? Lightgrey and this type of red are not really good distinguishable

Reply:
Mmh. I like light grey and the current red-ish. Which colours do you have in mind? You can change this already by defining another `param$col` and `param$col_samples`.

Reply:
Maybe we could exchange the lightgrey with a light yellow greyish colour?

Reply:
I can introduce yet another colour parameter, so there can be a personal choice of colour for the background.

Reply:
Is it ok for you, if I introduce the new background colour once I work on the custom theme? Would be next on my todo list, after this.

Reply:
how about just do: `param$col = c("lightgrey","red")`

Reply:
`param$col` has to be a single colour, given the way it is used at the moment. If you just want to switch from my pink-ish colour to red, this is already possible with this parameter. If you prefer to change the background colour as well, we can introduce `param$col_background`, for example.